### DISCUSSION TITLE
[RFC] 032 - indexedDB 持久化请求

### DISCUSSION THREAD

**Discussion by arvinxx** [lobehub/lobehub#1685] [RFCs]
## 背景

有几个用户遇到了聊着聊着，浏览器中的数据全没了的情况：

- #1666
- #904
- #1883
- #1463
- #2863 

通过调研发现，这个问题是在用户设备空间不够的情况下，浏览器会自动清理 indexedDB 造成的。

用户侧的解决方案是：

**把系统空间保持预留到 10% 的可用空间，或者2G空间**

参考资料
- https://developer.mozilla.org/zh-CN/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#%E6%95%B0%E6%8D%AE%E5%82%A8%E5%AD%98%E9%99%90%E5%88%B6

![image](https://github.com/lobehub/lobe-chat/assets/28616219/64b625ae-aec9-48c1-b924-044a773b24ad)

## 设计思路

产品层面我们需要给一个机制来保证这个问题不再出现

在上述参考资料中有相关介绍，可以提供相应的解决方案

## 进展

- https://github.com/lobehub/lobe-chat/pull/7089

**Comment by Wxh16144** [comment]
是不是可以套 Electron + SQLite 方案了🤔

**Reply by arvinxx** [reply]
不是，纯技术方案申请 indexedDB 的持久化应该就好了

**Comment by skywolf123** [comment]
这种被自动清理的有手段可以找回吗？

**Reply by sxjeru** [reply]
可以尝试用数据恢复软件扫描以下路径：
`%userdata%\AppData\Local\Google\Chrome\User Data\Default\IndexedDB`

**Reply by arvinxx** [reply]
> 可以尝试用数据恢复软件扫描以下路径：
> %userdata%\AppData\Local\Google\Chrome\User Data\Default\IndexedDB

+1

**Comment by countree103103** [comment]
这个问题在新版本完善了吗?
